runtime/thunk/thunk.ts

```diff
@@ -124,6 +125,34 @@ function doFindAndExecutePackFunction<T extends FormulaSpecification>({
             return parentFormula.execute(params as any, executionContext);
           }
           break;
+        case MetadataFormulaType.CellAutocomplete:
+          const syncTable = syncTables?.find(table => table.name === formulaSpec.syncTableName);
+          const autocompleteFn = ensureExists(syncTable?.autocompleteCell);
+          const propertyValues = {};
+
+          const cacheKeysUsed: string[] = [];
+
+          for (const [key, value] of Object.entries(formulaSpec.propertyValues)) {
+            Object.defineProperty(propertyValues, key, {
+              get() {
+                cacheKeysUsed.push(key);
+                return value;
+              },
+            });
+          }
+
+          const cellAutocompleteCxecutionContext: CellAutocompleteExecutionContext = {
+            ...executionContext,
+            propertyName: formulaSpec.propertyName,
+            propertyValues,
+          };
+          const result = await autocompleteFn.execute(params as any, cellAutocompleteCxecutionContext);
+          return {
+            result,
+            // TODO(dweitzman): Keys used should be an object or array, not a string
+            cacheKey: `keys used: ${cacheKeysUsed.join(',')}`,
```

Comment:
I'm having trouble understanding why we need to return cache keys here. Isn't it up to the client to keep track of what permutations of request params have already been used and cache them locally?

Reply:
The client passes all parameters to the pack, with no way of knowing which parameters were actually used unless the pack returns that information back

An alternative approach would be to have the pack formula pre-declare somehow in the schema which properties might influence the autocomplete of each other property. With that approach people would have to manually update their dependencies, and mistakes would lead to bugs.